src/raster/r.random.walk/r.random.walk.py

```diff
@@ -0,0 +1,475 @@
+#               Public License (>=v2). Read the file COPYING that
+#               comes with GRASS for details.
+#
+#############################################################################
+
+#%module
+#% description: Performs a 2D random walk inside the computational region and returns the resulting walk.
+#% keyword: raster
+#% keyword: random
+#% keyword: walk
+#% keyword: surface
+#% keyword: parallel
+#%end
+
+#%flag
+#% key: avoid
+#% description: Preform a self-avoiding random walk.
+#% guisection: Parameters
+#%end
+
+#%flag
+#% key: seed
+#% description: Generate random seed (result is non-deterministic).
+#%end
+
+#%flag
+#% key: tpath
+#% description: Each walker starts from the same point.
+#%end
+
+#%option G_OPT_R_OUTPUT
+#%end
+
+#%option
+#% key: steps
+#% type: integer
+#% required: no
+#% multiple: no
+#% description: How many steps to take during walk.
+#% answer: 100000
+#% guisection: Parameters
+#%end
+
+#%option
+#% key: directions
+#% type: string
+#% required: no
+#% multiple: no
+#% options: 4, 8
+#% description: How many directions should be used during walk.
+#% answer: 4
+#% guisection: Parameters
+#%end
+
+#%option G_OPT_MEMORYMB
+#%end
+
+#%option
+#% key: seed
+#% type: integer
+#% required: no
+#% multiple: no
+#% description: Seed for random number generator
+#%end
+
+#%option G_OPT_M_NPROCS
+#%end
+
+#%option
+#% key: nwalkers
+#% type: integer
+#% required: no
+#% multiple: no
+#% answer: 1
+#% description: Number of walkers.
+#%end
+
+import atexit
+import concurrent.futures
+import math
+import os
+import random
+import sys
+import time
+
+import grass.script as gs
+from grass.exceptions import CalledModuleError
+from grass.pygrass import raster
+from grass.pygrass.gis.region import Region
+from grass.script.core import gisenv
+
+# import numpy as np
+TMP_SMOOTH_RASTERS = []
+TMP_RASTERS = []
+PREFIX = "r_random_walk_temp_walk_"
+
+
+def cleanup():
+    if TMP_RASTERS:
+        gs.run_command(
+            "g.remove",
+            type="raster",
+            pattern="r_random_walk_temp_walk_*",
+            flags="f",
+            quiet=True,
+        )
+
+
+class GetOutOfLoop(Exception):
+    """
+    Throw to break out of nested loop.
+    """
+
+    pass
+
+
+## TODO: Rewrite this section to use differnt random distributions using values [0,1]
+def take_step(current_position, num_dir, black_list=None):
+    """
+    Calculates the next position of walker using either 4 or 8 possible directions.
+    :param list[row, column] current_position: A list with current row and column as integers.
+    :param int num_dir: The number of directions used in walk. Value must be either 4 or 8.
+    :param list[int] black_list: List of directions that the walker cannot take ranging from 0 - 7.
+    :return list[row, column] new_position
+    """
+    if black_list is None:
+        black_list = []
+
+    if num_dir not in [4, 8]:
+        raise ValueError(f"Unsupported num_dir Recieved: {num_dir}")
```

Comment:
```suggestion
        raise ValueError(f"Unsupported num_dir received: {num_dir}")
```

Reply:
Message wording: Plain `Unsupported num_dir: ...` should be enough. No need for "received" at the end, although I think Grassy the Hungry Cow uses that:

```
 ___________________________ 
| Bad grass input received! |
 --------------------------- 
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
_\|/__\|/__\|/_ ||     ||  _\|/__\|/_
```